### PR TITLE
Make plaintext inputs/weights configurable

### PR DESCRIPTION
**Pull request by mathias-nillion**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs change / refactoring / dependency upgrade

## Motivation and Context / Related issue
Makes plaintext weights and/or inputs a configurable variable when you convert a PyTorch model

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: curl/nn/module.py [hunk 1]
```diff
@@ -524,7 +525,7 @@ def forward_function(*args, **kwargs):
                         "Please make sure you feed your model CrypTensors when needed.",
                         DeprecationWarning,
                     )
-            elif not self.encrypted:
+            elif not self.encrypted and not self.SUPPORTS_PLAINTEXT_WEIGHTS:
                 if any(isinstance(arg, curl.CrypTensor) for arg in args):
```
Comment (jimouris) [inline curl/nn/module.py:529]: Is skipping this error here sufficient? Does it work or does it need more changes?

Comment (mathias-nillion) [reply curl/nn/module.py:529]: Tested it & seems to work indeed.
Can verify by running:
```
from torch import nn
fc = nn.Linear(100, 2)
private_model = curl.nn.from_pytorch(fc, torch.empty((1, 100)), allow_plaintext_weights=True)
private_model.eval()
output_enc = private_model(curl.cryptensor(torch.ones(1,100)))
output_enc.get_plain_text()
```
On a meta note: is this a good time to create testing + CI/CD?

**Review by jimouris — APPROVED** [review]

File: curl/nn/module.py (hunk 1 above)
Comment (jimouris) [reply curl/nn/module.py:529]: > On a meta note: is this a good time to create testing + CI/CD?

Yeah, that'd be nice but we'd have to be careful which tests to run as some take a while.

**Comment by jcabrero** [comment]
I didn't have the time to check, but are we sure that the operation being performed is `share <op> plaintext`?

**Pull request closed without merging**